src/Microsoft.AspNetCore.Razor.Tools/EliminateMethodBodyPass.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Razor.Tools
```

Comment:
hmmmm Does this need to be in this assembly?

If you're inserting this pass based on an option, I'd rather see the option set by the tool and then have this pass on all of the time, and no op when the option is set. Think about Razor.Language as an API

Reply:
Moved in to the main assembly and put it behind a CodeGenerationOption